### PR TITLE
FAI-11598: Commits Source Full Sync

### PR DESCRIPTION
**Pull request by matiaslcoulougian**
## Description
Source stream for commits. Following PRS: Converter and incremental sync support.

## Type of change
- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change

## Related issues

> Fix [#1]() 

## Migration notes

> Describe migration notes if any

## Extra info

> Add any additional information

### CODE REVIEW

File: sources/github-source/src/streams/faros_commits.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    const org = streamSlice?.org;
+    const repo = streamSlice?.repo;
+    const defaultBranch = streamSlice?.defaultBranch;
+    const github = await GitHub.instance(this.config, this.logger);
+    const commits = await github.getCommits(org, repo, defaultBranch);
+    for (const commit of commits) {
+      yield commit;
+    }
+  }
+}
```
[the remark this answers is not in the document]
Comment (matiaslcoulougian) [reply sources/github-source/src/streams/faros_commits.ts:35]: I will add it in a following PR. I need some of the things that are being added in https://github.com/faros-ai/airbyte-connectors/pull/1578

File: faros-airbyte-common/resources/github/queries/commits-query.gql [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+                  login
+                  url
+                }
+              }
+              committer {
+                date
+              }
+              additions
+              deletions
+              changedFilesIfAvailable
```
[the remark this answers is not in the document]
Comment (matiaslcoulougian) [reply faros-airbyte-common/resources/github/queries/commits-query.gql:30]: Will add it. Thought we were not supporting this any more. `changedFilesIfAvailable` was introduced in enterprise server version [3.8](https://docs.github.com/en/enterprise-server@3.8/graphql/reference/objects#:~:text=01%2D01%20UTC.-,changedFilesIfAvailable,-(Int)), and all previous versions are not supported any more

File: sources/github-source/src/github.ts [hunk 3]
```diff
@@ -139,6 +142,43 @@ export abstract class GitHub {
...
+    const iter = this.octokit(org).graphql.paginate.iterator<CommitsQuery>(
+      COMMITS_QUERY,
+      {
+        owner: org,
+        repo,
+        branch,
+        page_size: PAGE_SIZE,
+      }
+    );
+    for await (const res of iter) {
```
[the remark this answers is not in the document]
Comment (matiaslcoulougian) [reply sources/github-source/src/github.ts:174]: In this case it is not necessary the collection, left it by mistake. Will remove it

File: faros-airbyte-common/resources/github/queries/commits-query.gql (hunk 2 above)
[the remark this answers is not in the document]
Comment (chalenge) [reply faros-airbyte-common/resources/github/queries/commits-query.gql:30]: Well customers can stay on the old versions if they choose, they do not always upgrade.

**Comment by chalenge** [comment]
@matiaslcoulougian can we add the perms need for this stream

**Comment by matiaslcoulougian** [comment]
> @matiaslcoulougian can we add the perms need for this stream

Do you suggest adding the permissions table in the source's README?

File: faros-airbyte-common/resources/github/fragments/commit-fields.gql [hunk 4]
```diff
@@ -0,0 +1,20 @@
+fragment CommitFields on Commit {
```
Comment (chalenge) [inline faros-airbyte-common/resources/github/fragments/commit-fields.gql:1]: Nice 🚀

File: destinations/airbyte-faros-destination/src/converters/github/faros_commits.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+export class FarosCommits extends GitHubConverter {
+  readonly destinationModels: ReadonlyArray<DestinationModel> = ['vcs_Commit'];
+
+  async convert(
+    record: AirbyteRecord,
+    ctx: StreamContext
+  ): Promise<ReadonlyArray<DestinationRecord>> {
+    const commit = record.record.data as Commit;
+    const source = this.streamName.source;
+    return [];
```
Comment (chalenge) [inline destinations/airbyte-faros-destination/src/converters/github/faros_commits.ts:16]: `return []`??

File: sources/github-source/README.md [hunk 6]
```diff
@@ -75,4 +75,13 @@ We use [lerna](https://lerna.js.org/) to manage dependencies that are shared by
...
 connectors, such as linting/formatting tools, should go in the root
-`package.json`.
\ No newline at end of file
+`package.json`.
+
+### GitHub Required Permissions per Stream
+
+| Stream  | Classic               | Fine-grained*                                                     |
+|---------|-----------------------|-------------------------------------------------------------------|
+| Commits | repo:status, read:org | Repository Contents, Repository Metadata, Organization Members    |
```
Comment (chalenge) [inline sources/github-source/README.md:84]: mostly curious, why does this need `Organization Members`?

Comment (matiaslcoulougian) [reply sources/github-source/README.md:84]: Thanks for the question. Thought it was needed for organization listing. Double-checked and the request does not require any permission. [Doc](https://docs.github.com/en/rest/orgs/orgs?apiVersion=2022-11-28#list-organizations-for-the-authenticated-user--fine-grained-access-tokens:~:text=personal%20access%20tokens-,The%20fine%2Dgrained%20token%20does%20not%20require%20any%20permissions.,-Parameters%20for%20%22List)

**Review by chalenge — APPROVED** [review]
lgtm

**Pull request merged**